### ISSUE TITLE
TypeError: process_events() got an unexpected keyword argument 'lib'

### ISSUE THREAD

**Issue by wazabees** [adammillerio/beets-copyartifacts#2]
Hello!
I'm not sure if you're still working om this plugin, but I figured I'd mention a potential bug.

Output from beet version:
```
beets version 1.5.0
Python version 3.8.5
plugins: copyartifacts, edit, fetchart, importadded, importfeeds, info, inline, mbcollection, mbsync, mpdstats, mpdupdate, originquery, scrub, smartplaylist, zero
```
And the error message:
```
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
copyartifacts: Ignored files:
copyartifacts:    cover.jpg
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.8/site-packages/beets/plugins.py", line 145, in wrapper
    return func(*args, **kwargs)
TypeError: process_events() got an unexpected keyword argument 'lib'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/.local/bin/beet", line 8, in <module>
    sys.exit(main())
  File "/home/user/.local/lib/python3.8/site-packages/beets/ui/__init__.py", line 1269, in main
    _raw_main(args)
  File "/home/user/.local/lib/python3.8/site-packages/beets/ui/__init__.py", line 1258, in _raw_main
    plugins.send('cli_exit', lib=lib)
  File "/home/user/.local/lib/python3.8/site-packages/beets/plugins.py", line 492, in send
    result = handler(**arguments)
  File "/home/user/.local/lib/python3.8/site-packages/beets/plugins.py", line 151, in wrapper
    return func(*args, **kwargs)
  File "/home/user/.local/lib/python3.8/site-packages/beetsplug/copyartifacts.py", line 127, in process_events
    self.process_artifacts(item['files'], item['mapping'], False)
  File "/home/user/.local/lib/python3.8/site-packages/beetsplug/copyartifacts.py", line 161, in process_artifacts
    dest_file = beets.util.unique_path(dest_file)
  File "/home/user/.local/lib/python3.8/site-packages/beets/util/__init__.py", line 561, in unique_path
    match = re.search(br'\.(\d)+$', base)
  File "/usr/lib/python3.8/re.py", line 201, in search
    return _compile(pattern, flags).search(string)
TypeError: cannot use a bytes pattern on a string-like object
```

**Comment by Stunner** [comment]
I get this same error... all audio files are moved but the albumart (which I have configured to also move) was left behind:

```
Traceback (most recent call last):
  File "/volume1/@appstore/beets/env/lib/python3.7/site-packages/beets/plugins.py", line 143, in wrapper
    return func(*args, **kwargs)
TypeError: process_events() got an unexpected keyword argument 'lib'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/beet", line 10, in <module>
    sys.exit(main())
  File "/volume1/@appstore/beets/env/lib/python3.7/site-packages/beets/ui/__init__.py", line 1266, in main
    _raw_main(args)
  File "/volume1/@appstore/beets/env/lib/python3.7/site-packages/beets/ui/__init__.py", line 1255, in _raw_main
    plugins.send('cli_exit', lib=lib)
  File "/volume1/@appstore/beets/env/lib/python3.7/site-packages/beets/plugins.py", line 489, in send
    result = handler(**arguments)
  File "/volume1/@appstore/beets/env/lib/python3.7/site-packages/beets/plugins.py", line 149, in wrapper
    return func(*args, **kwargs)
  File "/var/services/homes/aaron_admin/.local/lib/python3.7/site-packages/beetsplug/copyartifacts.py", line 127, in process_events
    self.process_artifacts(item['files'], item['mapping'], False)
  File "/var/services/homes/aaron_admin/.local/lib/python3.7/site-packages/beetsplug/copyartifacts.py", line 161, in process_artifacts
    dest_file = beets.util.unique_path(dest_file)
  File "/volume1/@appstore/beets/env/lib/python3.7/site-packages/beets/util/__init__.py", line 550, in unique_path
    match = re.search(br'\.(\d)+$', base)
  File "/usr/local/python3/lib/python3.7/re.py", line 183, in search
    return _compile(pattern, flags).search(string)
TypeError: cannot use a bytes pattern on a string-like object
```

**Comment by DOFfactory** [comment]
Same here, was a happy user of the plugin but it ends in errors...

**Comment by s1shed** [comment]
I made this change with the original, unforked plugin in order to get Python 3 support. 
 
```diff
diff --git a/beetsplug/copyartifacts.py b/beetsplug/copyartifacts.py
index 2e9593f..f48300f 100644
--- a/beetsplug/copyartifacts.py
+++ b/beetsplug/copyartifacts.py
@@ -158,7 +158,7 @@ class CopyArtifactsPlugin(BeetsPlugin):
                 ignored_files.append(source_file)
                 continue
 
-            dest_file = beets.util.unique_path(dest_file)
+            dest_file = beets.util.unique_path(dest_file.encode('utf8'))
             beets.util.mkdirall(dest_file)
             dest_file = beets.util.bytestring_path(dest_file)
 ```
 

Perhaps this will fix the crash reported in this issue. I have not been able to reproduce the crash with this applied.

**Comment by adammillerio** [comment]
This should be resolved with #4 and is available in the latest release 0.1.4 of beets-copyartifacts3.
